core/commands/swarm.go

```diff
@@ -447,6 +463,51 @@ func (ci connInfos) Swap(i, j int) {
 	ci.Peers[i], ci.Peers[j] = ci.Peers[j], ci.Peers[i]
 }
 
+func (ci *connInfo) identifyPeer(ps pstore.Peerstore, p peer.ID) (*IdOutput, error) {
+	info := new(IdOutput)
+	keyEnc, err := ke.KeyEncoderFromString(ke.OptionIPNSBase.Name())
+	if err != nil {
+		return nil, err
+	}
+	info.ID = keyEnc.FormatID(p)
+
+	if pk := ps.PubKey(p); pk != nil {
+		pkb, err := ic.MarshalPublicKey(pk)
+		if err != nil {
+			return nil, err
+		}
+		info.PublicKey = base64.StdEncoding.EncodeToString(pkb)
+	}
+
+	addrInfo := ps.PeerInfo(p)
+	addrs, err := peer.AddrInfoToP2pAddrs(&addrInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, a := range addrs {
+		info.Addresses = append(info.Addresses, a.String())
+	}
+	sort.Strings(info.Addresses)
+
+	protocols, _ := ps.GetProtocols(p) // don't care about errors here.
+	info.Protocols = append(info.Protocols, protocols...)
+	sort.Slice(info.Protocols, func(i, j int) bool { return info.Protocols[i] < info.Protocols[j] })
```

Comment:
```suggestion
	if protocols, err := ps.GetProtocols(p); err == nil {
		info.Protocols = append(info.Protocols, protocols...)
		sort.Slice(info.Protocols, func(i, j int) bool { return info.Protocols[i] < info.Protocols[j] })
	}
```

I could be wrong, but I don't belive you can get partial protocol data if an error happens here.

Reply:
That's my first PR on this repo so I actually don't know, thank you for the suggestion, I'll add it to the code

Reply:
What you wrote might be right, as a sanity thing check errors before using the other return values (unless this is explicitelly specified to be valid when `err != nil`, random example `io.Reader`'s `int` is valid even if an error happend)